File: java/factory/Diners.java
package factory;

public class Diners extends LocalPaymentMethod {

    public Diners(String cardHolder, String cardNumber, String cvv, String expiryDate) {
        super(cardHolder, cardNumber, cvv, expiryDate);
    }

    @Override
    public void authorizePayment() {
        System.out.println("Authorizing Diners card....");
    }

    @Override
    public void startMoneyTransfer() {
        System.out.println("Start money transferring for a Diners card....");
    }

    @Override
    public void calculatePaymentFees(double amount) {
        System.out.println("Calculating Diners card payment fees....");
    }
}

File: typescript/factory/Diners.ts
import { LocalPaymentMethod } from "./local-payment-method";

export class Diners extends LocalPaymentMethod {
    constructor(
      cardHolder: string,
      cardNumber: string,
      cvv: string,
      expiryDate: string
    ) {
      super(cardHolder, cardNumber, cvv, expiryDate);
    }
  
    authorizePayment(): void {
      console.log("Authorizing Diners card....");
    }
  
    startMoneyTransfer(): void {
      console.log("Start money transferring for a Diners card....");
    }
  
    calculatePaymentFees(amount: number): void {
      console.log("Calculating Diners card payment fees....");
    }
  }
  